kfs-core/src/main/java/edu/arizona/kfs/tax/document/IncomeTypeHandler.java

```diff
@@ -0,0 +1,576 @@
+		String retval = getTaxParameterHelperService().getOverridePaymentType(vendorHeader, overridePaymentTypeCodeMap);
+		if (StringUtils.isBlank(retval)) {
+			String finObjectCode = acctline.getFinancialObjectCode();
+
+			if (StringUtils.isNotBlank(finObjectCode) && objectCodeMap.containsKey(finObjectCode)) {
+				retval = objectCodeMap.get(finObjectCode);
+			} else {
+				// Set payment type to type 7 by default
+				retval = TaxConstants.AMOUNT_CODE_7;
+			}
+		}
+
+		return retval;
+	}
+
+	/**
+	 * This method checks to see if the current document is cancelled or approved - in those cases we will
+	 * not allow the income types to pre-populate
+	 * 
+	 * @return
+	 */
+	public boolean isEditableRouteStatus() {
+		String routeStatus = incomeTypeDocumentContainer.getRouteStatus();
+
+		return (KewApiConstants.ROUTE_HEADER_INITIATED_CD.equals(routeStatus) || KewApiConstants.ROUTE_HEADER_SAVED_CD.equals(routeStatus) || KewApiConstants.ROUTE_HEADER_ENROUTE_CD.equals(routeStatus));
+	}
+
+	/**
+	 * This method loops over the current DocumentIncomeType list on this document and sets the amount to 0
+	 */
+	private void zeroOutDocumentIncomeTypes() {
+		for (T1 dit : incomeTypeDocumentContainer.getIncomeTypes()) {
+			dit.setAmount(new KualiDecimal(0));
+		}
+	}
+
+	/**
+	 * This looks for a DocumentIncomeType in the current document incomeTypes Collection that
+	 * matches our AccountingLine. If it does not find a match and this accounting line is a valid 1099
+	 * type then we create a new DocumentIncomeType and return it
+	 * 
+	 * @param acctline
+	 * @param box
+	 * @return
+	 */
+	private T1 getDocumentIncomeType(AccountingLine acctline, String box) {
+		T1 retval = null;
+
+		if (LOG.isDebugEnabled()) {
+			LOG.debug("financialObjectCode:" + acctline.getFinancialObjectCode());
+		}
+
+		// loop over current DocumentIncomeTypes and see if we can find
+		// a match for AccountingLine
+		for (T1 dit : incomeTypeDocumentContainer.getIncomeTypes()) {
+			if (box.equals(dit.getIncomeType().getIncomeTypeBox())) {
+				retval = dit;
+				break;
+			}
+		}
+
+		if (LOG.isDebugEnabled()) {
+			if (retval == null) {
+				LOG.debug("did not find income type in map");
+			} else {
+				LOG.debug("found income type " + retval.getIncomeTypeCode() + " in map");
+			}
+		}
+
+		// if DocumentIncomeType does not exist - create one
+		if (retval == null) {
+			IncomeType ic = incomeTypeMap.get(box);
+			retval = getInitializedDocumentIncomeType();
+			retval.setChart(acctline.getChart());
+			retval.setChartOfAccountsCode(acctline.getChartOfAccountsCode());
+			retval.setIncomeType(ic);
+			retval.setIncomeTypeCode(ic.getIncomeTypeCode());
+			retval.setLineNumber(getNextIncomeTypeLineNumber());
+
+			// add the new DocumentIncomeType to the list
+			incomeTypeDocumentContainer.getIncomeTypes().add(retval);
+		}
+
+		return retval;
+	}
+
+	private String get1099Box(VendorHeader vendorHeader, String financialObjectCode) {

```

Comment:
Each of these get1099Box methods are only used in 1 place. A comment has been added to both each, indicating what methods use them.